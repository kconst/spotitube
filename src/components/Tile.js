import React, { PropTypes, Component } from 'react';

export default class Tile extends Component {
  render() {
    const { name, description, images } = this.props;

    return (
      <li className="Tile">
        <img src={ images[0] && images[0].url } alt={ name }/>
        <h2>
          { name }
        </h2>

        <p>
          { description }
        </p>
      </li>
    )
  }
}

Tile.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};