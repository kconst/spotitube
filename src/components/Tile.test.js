// Link.react-test.js
import React from 'react';
import Tile from './Tile';
// import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

test('basic rendering', () => {
  const jsx = <Tile image="http://iscale.iheart.com/catalog/artist/58269?ops=fit(250,0)" name="Metallica" description="they a band"></Tile>,
    //   component = renderer.create(jsx),
      tile = shallow(jsx);
      
      // snapshot check
    // let tree = component.toJSON();
    // expect(tree).toMatchSnapshot();

    expect(tile.find('h2').text()).toEqual('Metallica');
});