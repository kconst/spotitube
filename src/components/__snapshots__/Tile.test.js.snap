exports[`test snapshot comparison 1`] = `
<li
  className="Tile"
  onClick={[Function]}>
  <img
    alt="Trivium"
    src="https://i.scdn.co/image/5194f3860bd53aa613edcfc5ef09a995b599f049" />
  <h2>
    Trivium
  </h2>
</li>
`;
